src/toniarts/openkeeper/world/WorldState.java

```diff
@@ -188,6 +191,26 @@ private void initPlayerCreatures() {
 
                 @Override
                 public void onSpawn(CreatureControl creature) {
+                    if (player.getId() == stateManager.getState(PlayerState.class).getPlayerId() && player.getCreatureControl().getTypeCount(creature.getCreature()) == 0) {

```

Comment:
This will do for now, but in the long run this doesn't work. As they get killed or sacked. Easy enough to track this per level, but how about on campaign (it doesn't show this on Goblins every time)? Is there some parameters on the level data?

No need to solve this now though, this is great just as it is. Just starting a conversation :)